operations/node_management/Defragment_NID_Numbering.md

```diff
@@ -0,0 +1,337 @@
+# Defragment NID Numbering
+
+This procedure will rearrange NIDs for specified compute nodes to create a numerically (NID) and lexicographically (xname) contiguous block of NIDs at the specified start point.
+
+It is recommended that the system be taken down for maintenance while performing this procedure.
+
+This procedure should only be performed if absolutely required. Some reasons for needing to perform this procedure include:
+
+1. Compute nodes were added to SLS with incorrect NID numbering, missing node entries, and/or extra node entries.
+
+2. Compute nodes were permanently moved, removed, or reprovisioned and there is a desire to remove NID numbering gaps.
+
+The example in this procedure removes NID gaps from 2 cabinets of compute nodes that were a result of incorrect numbering in SLS.
+
+## Topics
+
+* [Prerequisites](#prerequisites)
+* [Defragment NID Script Functionality and Limitations](#defragment-nid-script-functionality-and-limitations)
+* [NID Defragmentation Procedure](#nid-defragmentation-procedure)
+  * [Step 1: Run defragment_nids.py](#step-1-run-defragment_nids.py)
+  * [Step 2: Perform Reload of DVS/LNet Service](#step-2-perform-reload-of-dvslnet-service)
+* [Troubleshooting](#Troubleshooting)
+  * [Discovery Errors](#discovery-errors)
+  * [Invalid NID Range](#invalid-nid-range)
+
+## Prerequisites
+
+* Hardware State Manager's (HSM) inventory data is up to date (Hardware has not changed since the previous discovery).
+* Chassis-level entries exist in the System Layout Service (SLS) and are correct for all Mountain and Hill chassis.
+* All compute nodes are powered off.
+
+## Defragment NID Script Functionality and Limitations
+
+In the process of defragmenting NIDs, the defragment_nids.py script will:
+
+* Update the NID numbers for compute node entries in HSM.
+* Update/create SLS compute node entries with correct NID numbering and aliases.
+* Remove node entries from HSM that were not previously removed due to missed blade swap procedure steps. This includes associated entries under `/State/Components`, `/Inventory/ComponentEndpoints`, `/Inventory/Hardware`, and `/Inventory/EthernetInterfaces`.
+* Remove SLS node entries with conflicting NIDs that are not in HSM.
+* Correct the _Class_ designation of compute nodes in HSM.
+
+Limitations of the defragment_nids.py script:
+
+* Only affects compute nodes.
+* HSM node entries are only removed if the "leftover" nodes are of a different model than the existing nodes in the same slot.
+* SLS node entries that don't exist in HSM are only removed if their NID falls within the specified NID block.
+
+## NID Defragmentation Procedure
+
+### Step 1: Run defragment_nids.py
+
+1. Choose the starting NID for the NID block (e.g., 1000).
+
+```bash
```

Comment:
These fenced blocks should also be tabbed over, this will render the blocks correctly as part of the listed step.

Reply:
fixed